plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java

```diff
@@ -235,14 +237,17 @@ protected void refreshFromMetadataLocation(String newLocation)
                             .withMaxRetries(20)
                             .withBackoff(100, 5000, MILLIS, 4.0)
                             .withMaxDuration(Duration.ofMinutes(10))
-                            .abortOn(AbstractIcebergTableOperations::isNotFoundException)
+                            .abortOn(AbstractIcebergTableOperations::shouldBeAborted)
```

Comment:
can inline `shouldBeAborted` since we need to handle them explicitly in `catch` anyway